cogs/cipher.py

```diff
@@ -0,0 +1,129 @@
+import base64
+import hashlib
+import re
+import string
+
+from discord.ext import commands
+
+import cogs.utils.parser as parser
+from assets.context import Command, argument, example, usage_info
+
+
+class Ciphers(commands.Cog):
+    character_set = [string.ascii_lowercase, string.ascii_uppercase, string.digits]
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def cog_check(self, ctx):
+        if self.bot.is_ready():
+            return True
+        else:
+            return False
+
+    @commands.command(cls=Command,
+                      description="Shifts the letters within a message",
+                      syntax=[argument(True, "Shift defaults to 3"), argument(True, "Message")],
+                      usage_information=[
+                          usage_info("", "If a shift value is not found at the start "
+                                         "of the message it would try and find it at the end")
+                      ],
+                      examples=[argument("Hello", "Returns Khoor"), argument("10 Hello", "Returns Rovvy")])
+    async def shift(self, ctx, *, message):
+        message = message.split()
+        shift = await parser.back_front_parser(message, is_list=True)
+
+        for pos_neg in ["-", ""]:
+            if re.findall(f"{pos_neg}\d+", " ".join(shift)):
+                shift = re.findall(f"{pos_neg}\d+", " ".join(shift))[0]
+                break
+        if not shift or isinstance(shift, list) or shift not in message:
+            shift = 3
+            message = " ".join(message)
+        else:
+            location = message.index(shift)
+            del message[location]
+            shift = int(shift)
+            message = " ".join(message)
+
+        if not message:
+            return await ctx.send(f"I need some text to shift by {shift} places!")
+
+        # If the shift is higher than 25 then this would convert the shift to a value that is lower than 25
+        if shift >= 26:
+            while shift >= 26:
+                shift %= 26
+        elif shift <= -26:
+            while shift <= -26:
+                shift %= -26
```

Comment:
```suggestion
        shift %= 26
```
The modulo operation always returns something positive. It's the rest in the Euclidian division